### PR TITLE
chore: Remove useless clients when possible (AR-2375)

### PR DESCRIPTION
**Pull request by borichellow**
# What's new in this PR?

When user Persist OtherUser Clients, it also removes clients that are not in the list. 
This will increase performance a bit while sending messages.

### Issues

PersistOtherUserClientsUsecase was only saving the new Clients into DB and does nothing on Clients that do not exist anymore. So when user sends the messages to that OtherUser, app crypts message for not existed Clients too.

### Causes (Optional)

It was implemented like that before 🤷‍♂️

### Solutions

Make `PersistOtherUserClientsUsecase` not only save new Clients but remove redundant too

### CODE REVIEW

File: persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/client/ClientDAO.kt [hunk 1]
```diff
@@ -18,4 +18,5 @@ interface ClientDAO {
     suspend fun deleteClientsOfUserByQualifiedID(qualifiedID: QualifiedIDEntity)
     suspend fun deleteClient(userId: QualifiedIDEntity, clientId: String)
     suspend fun getClientsOfConversation(id: QualifiedIDEntity): Map<QualifiedIDEntity, List<Client>>
+    suspend fun insertClientsAndRemoveRedundant(qualifiedID: QualifiedIDEntity, clients: List<Client>)
```
Comment (MohamadJaara) [inline persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/client/ClientDAO.kt:21]: Missing unit test

**Review by MohamadJaara — CHANGES_REQUESTED** [review]
great job 🚀, only missing tests for `insertClientsAndRemoveRedundant`

File: persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/client/ClientDAO.kt (hunk 1 above)
Comment (borichellow) [reply persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/client/ClientDAO.kt:21]: unitTests are missed for the whole `ClientDAOImpl` ;P

File: persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/client/ClientDAOImpl.kt [hunk 2]
```diff
@@ -32,6 +32,14 @@ internal class ClientDAOImpl internal constructor(
         }
     }
 
+    override suspend fun insertClientsAndRemoveRedundant(qualifiedID: QualifiedIDEntity, clients: List<Client>) =
+        clientsQueries.transaction {
+            clients.forEach { client ->
+                clientsQueries.insertClient(client.userId, client.id, client.deviceType)
+            }
+            clientsQueries.deleteClientsOfUserExcept(qualifiedID, clients.map { it.id })
+        }
```
Comment (vitorhugods) [inline persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/client/ClientDAOImpl.kt:41]: Considering this is done in a loop, using [Grouped Statements](https://cashapp.github.io/sqldelight/multiplatform_sqlite/grouping_statements/) is significantly better than just a transaction, performance-wise.

**Review by vitorhugods — APPROVED** [review]

**Review by MohamadJaara — APPROVED** [review]
LGTM 🚀

**Pull request merged**